wordpress-mobile/WordPress-Android
# UC: Comment List (feature branch PR)

Fixes #15101

This is feature branch PR, with all the commits (apart from correct flux-c has) PR's reviewed.

Flux-C counterpart is [here](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2081).

New comment list is utilizing modern android development practices, and is utilizing Room with Flow.
Legacy comment functionality is working through adapter.

To test:
- Test legacy comment functionality everywhere StoreAdapter is used (eg. Notifications).
- Check interactions with a comment list - moderation and paging (wpcom and self-hosted).
- Confirm that the feature flag is working.


## Regression Notes
1. Potential unintended areas of impact
- Legacy comment functionality that is utilizing StoreAdapter might not work.
- Something might potentially not work in new comment list.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- We added a bunch of tests to make sure everything works as expected.
- Manual testing.

3. What automated tests I added (or what prevented me from doing so)
- Unit tests.

PR submission checklist:
